src/osd/CMakeLists.txt

```diff
@@ -2,6 +2,13 @@ set(osdc_osd_srcs
   ${CMAKE_SOURCE_DIR}/src/osdc/Objecter.cc
   ${CMAKE_SOURCE_DIR}/src/osdc/Striper.cc)
 
+if(WITH_OSD_INSTRUMENT_FUNCTIONS)
+  set(GCC_C_FLAGS "-finstrument-functions")
+  set(GCC_C_FLAGS "${GCC_C_FLAGS} -finstrument-functions-exclude-function-list=_mm_loadu_si128,_mm_cmpeq_epi32,_mm_movemask_epi8")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_C_FLAGS}")
```

Comment:
@cbodley I hadn't checked that, good point! I'm not too familiar with clang, but it seems -finstrument-functions is supported and behaves the same way based on quick runs I did. However, it doesn't support -finstrument-functions-exclude-function-list yet. Should I add make a check for GCC around this whole WITH_OSD_INSTRUMENT_FUNCTIONS config?